src/NHibernate/Dialect/SapSQLAnywhere17Dialect.cs

```diff
@@ -20,35 +20,30 @@ namespace NHibernate.Dialect
 	///	</listheader>
 	///	<item>
 	///		<term>connection.driver_class</term>
-	///		<description><see cref="NHibernate.Driver.SybaseSQLAnywhere17Driver" /></description>
+	///		<description><see cref="NHibernate.Driver.SapSQLAnywhere17Driver" /></description>
 	///	</item>
 	///	<item>
 	///		<term>prepare_sql</term>
 	///		<description><see langword="false" /></description>
 	///	</item>
 	/// </list>
 	/// </remarks>
-	public class SybaseSQLAnywhere17Dialect : SybaseSQLAnywhere12Dialect
+	public class SapSQLAnywhere17Dialect : SybaseSQLAnywhere12Dialect
 	{
-		public SybaseSQLAnywhere17Dialect()
+		public SapSQLAnywhere17Dialect()
 			: base()
 		{
 			DefaultProperties[Environment.ConnectionDriver] = "NHibernate.Driver.SybaseSQLAnywhere17Driver";
-			RegisterDateTimeTypeMappings();
-			RegisterKeywords();
 		}
-
-		new protected void RegisterKeywords()
+	
+		protected override void RegisterKeywords()
 		{
-			RegisterKeyword("near");
-			RegisterKeyword("limit");
-			RegisterKeyword("offset");
-			RegisterKeyword("datetimeoffset");
+			base.RegisterKeywords();
```

Comment:
So, indeed these keywords were already registered in the base dialect. Better remove completely the override then.